kafka/testhelpers.go

```diff
@@ -83,6 +89,11 @@ func testconfRead() bool {
 	testconf.Topic = defaulttestconfTopic
 	testconf.Brokers = ""
 
+	if testconf.Docker || testconf.Semaphore {
+		testconf.Brokers = defaulttestconfBrokers
+		return true
```

Comment:
We can't return true here, since the user might be using testconf.Docker = true in their local setup. In that case we still need to consider the value of the other fields.

We can perhaps do 
```
if testconf.Docker || testconf.Semaphore {
		testconf.Brokers = defaulttestconfBrokers
		if cf == nil {
                  return true

Reply:
Did not get you! If the docker is true then all the default values are taken and no other value is entertained even if the testconf has some different ones

Reply:
In that case, please close cf if it was opened. Currently, cf.Close() is never called corresponding to os.Open()

I suggest we do something like this:
``` 
       cf, err := os.Open("./testconf.json")
	if err != nil && !testconf.Docker && !testconf.Semaphore {
		fmt.Fprintf(os.Stderr, "%% testconf.json not found and docker compose not setup - ignoring test\n")
		return false
	}
	if err == nil {
	    defer cf.Close()
	}

Reply:
And remove line 103 which is currently closing cf.